src/compiler.js

```diff
@@ -163,30 +171,53 @@ import {${specifiers.map(
       main
         .variable(observer(reference))
         .define(reference, cellReferences, cellFunction);
-      main.variable(observer(cellName)).define(cellName, ["Generators", reference], (G, _) => G.input(_));
+      main
+        .variable(observer(cellName))
+        .define(cellName, ["Generators", reference], (G, _) => G.input(_));
     } else if (cell.id && cell.id.type === "MutableExpression") {
       const initialName = `initial ${cellName}`;
       const mutableName = `mutable ${cellName}`;
+      main.variable(null).define(initialName, cellReferences, cellFunction);
       main
-        .variable(null)
-        .define(initialName, cellReferences, cellFunction);
-      main.variable(observer(mutableName)).define(mutableName, ["Mutable", initialName], (M, _) => new M(_));
-      main.variable(observer(cellName)).define(cellName, [mutableName], _ => _.generator);
+        .variable(observer(mutableName))
+        .define(mutableName, ["Mutable", initialName], (M, _) => new M(_));
+      main
+        .variable(observer(cellName))
+        .define(cellName, [mutableName], _ => _.generator);
     } else {
       main
         .variable(observer(cellName))
         .define(cellName, cellReferences, cellFunction);
     }
   }
 };
-const createModuleDefintion = (m, resolveModule, resolveFileAttachments) => {
+const createModuleDefintion = async (
+  moduleObject,
+  resolveModule,
+  resolveFileAttachments
+) => {
+  const dependencyMap = new Map();
+
+  const importCells = moduleObject.cells.filter(
+    cell => cell.body.type === "ImportDeclaration"
+  );
+
+  const importCellsPromise = importCells.map(async cell => {
+    const fromModule = await resolveModule(cell.body.source.value);
+    dependencyMap.set(cell.body.source.value, fromModule);
+  });
+  await Promise.all(importCellsPromise);
+
   return async function define(runtime, observer) {
```

Comment:
I think `define` no longer needs to be `async`:
```suggestion
  return function define(runtime, observer) {
```
(see also change below)